src/PdfSharper/Pdf.AcroForms/PdfSignatureField.cs

```diff
@@ -112,7 +110,6 @@ public PdfSignatureField(PdfDocument document)
             sign.Elements.Add(Keys.M, new PdfDate(DateTime.Now));
 
             document._irefTable.Add(sign);
-            document._irefTable.Add(this);
```

Comment:
The code here is adding an already signed signature field to the document, I think we should change it up to only add the field to the document;   (no explicit calls to ireftable) especially now that we have support for adding fields externally.

Signing the document should be a separate process.